lib/bandit/io.ex

```diff
@@ -0,0 +1,55 @@
+defmodule Bandit.IO do
+  @moduledoc false
+
+  @doc """
+  Returns an iolist consisting of the first `length` bytes of the given list, along with the
+  remainder of the iolist beyond it as a tuple. Returns `{:error, :length_exceeded}` if there are
+  not `length` bytes remaining in the iolist.
+  """
+  @spec split_iodata(iodata(), pos_integer(), iodata(), iodata()) :: {iodata(), iodata()} | {:error, :length_exceeded}
+  def split_iodata(iodata, length, head \\ [], acc \\ [])
+  def split_iodata(iodata, 0, [] = head, []), do: {head, iodata}
+  def split_iodata([], _length, [], []), do: {:error, :length_exceeded}
+  def split_iodata("", _length, [], []), do: {:error, :length_exceeded}
+  def split_iodata([byte | _] = iodata, 0, head, []) when is_integer(byte), do: {head, iodata}
+  def split_iodata(_iodata, 0, head, [] = acc), do: {head, acc}
```

Comment:
This should return the iodata as second element and not the empty `acc`.
This also explains why the benchmark was that fast, since it dismissed a part of the data.

Reply:
Thanks for the review, that could explain the reduction in memory.